src/appshell/view/internal/splashscreen/newinstanceloadingscreenview.cpp

```diff
@@ -83,3 +92,23 @@ void NewInstanceLoadingScreenView::draw(QPainter* painter)
     QString elidedText = fontMetrics.elidedText(m_message, Qt::ElideMiddle, messageRectangle.width());
     painter->drawText(messageRectangle, Qt::AlignCenter | Qt::TextDontClip, elidedText);
 }
+
+void NewInstanceLoadingScreenView::drawRoundedRect(QPainter* painter, const QRectF& rect, const qreal radius,
+                                                   const QBrush& brush, const QPen& pen)
+{
+    painter->save();
+    painter->setPen(QPen(Qt::transparent, 0));
+    painter->setBrush(brush);
+
+    QPainterPath path;
+    path.addRoundedRect(rect, radius, radius);
+    painter->fillPath(path, brush);
```

Comment:
Why is this done using a QPainterPath and not just using `painter->drawRoundedRect`?